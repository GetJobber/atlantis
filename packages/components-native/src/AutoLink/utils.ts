import { EmailMatch, Match, PhoneMatch } from "autolinker";
import { Linking } from "react-native";
import { copyTextToClipboard } from "./clipboard";
import { I18nKeys, useAtlantisI18nValue } from "../hooks/useAtlantisI18n";

function hasPrefix(text: string, prefixes: string[]): boolean {
  return prefixes.some(prefix => text.includes(prefix));
}

export function shouldIgnoreURL(text: string, match: Match): boolean {
  const matchedText = match.getMatchedText().toLocaleLowerCase();
  const urlPrefixes = ["http://", "https://", "www."];
  const ignorePrefixes = ["file://", "ftp://"];
  const previousChar = text.charAt(match.getOffset() - 1);

  if (match.getType() === "url") {
    const hasUrlPrefix = hasPrefix(matchedText, urlPrefixes);
    const hasIgnorePrefix =
      hasPrefix(matchedText, ignorePrefixes) || previousChar === "/";

    return hasIgnorePrefix && !hasUrlPrefix;
  }

  return false;
}

export function getUrl(match: Match, immediateOpen = true): string {
  const matchType = match.getType();

  switch (matchType) {
    case "email":
      return immediateOpen
        ? `mailto:${encodeURIComponent((match as EmailMatch).getEmail())}`
        : (match as EmailMatch).getEmail();
    case "phone":
      return immediateOpen
        ? `tel:${(match as PhoneMatch).getNumber()}`
        : (match as PhoneMatch).getNumber();
    default:
      return match.getAnchorHref();
  }
}

export function onLongPressLink(
  match: Match,
  bottomTabsVisible: boolean,
  t: useAtlantisI18nValue["t"],
): void {
  const linkUrl = getUrl(match, false);

  const toastConfig = {
    message: t(getMessageKey(match)),
    bottomTabsVisible,
  };
  copyTextToClipboard(linkUrl, toastConfig);
}

export function onPressLink(match: Match): void {
  const linkUrl = getUrl(match);
  Linking.openURL(linkUrl);
}

function getMessageKey(match: Match): I18nKeys {
  switch (match.getType()) {
    case "email":
      return "AutoLink.emailCopied";
    case "phone":
      return "AutoLink.phoneCopied";
    default:
      return "AutoLink.urlCopied";
  }
}
