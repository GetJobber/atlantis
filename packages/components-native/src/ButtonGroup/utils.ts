import React, { useCallback } from "react";
import { useIntl } from "react-intl";
import { Alert } from "react-native";
import { messages } from "./messages";
import {
  ButtonGroupActionElement,
  ButtonGroupPrimaryActionElement,
} from "./types";
import {
  ButtonGroupActionProps,
  PrimaryAction,
  SecondaryAction,
} from "./ButtonGroupAction";
import { useAtlantisContext } from "../AtlantisContext/AtlantisContext";

interface UsePreventTapWhenOfflineShape {
  readonly handlePress: (
    callback: ButtonGroupActionProps["onPress"],
  ) => () => void;
}

/**
 * Determine if the onPress should be fired or an alert when the device is
 * online or offline
 */
export function usePreventTapWhenOffline(): UsePreventTapWhenOfflineShape {
  const { formatMessage } = useIntl();
  const { isOnline } = useAtlantisContext();

  const handlePress = useCallback(
    (callback: ButtonGroupActionProps["onPress"]) => {
      if (isOnline) return callback;

      return () =>
        Alert.alert(
          formatMessage(messages.unavailableNetworkTitle),
          formatMessage(messages.unavailableNetworkMessage),
        );
    },
    [formatMessage, isOnline],
  );

  return { handlePress };
}

interface GetActionShape {
  readonly primaryActions: ButtonGroupPrimaryActionElement[];
  readonly secondaryActions: ButtonGroupActionElement[];
}

/**
 * Separate the Primary and Secondary actions into 2 const
 */
export function getActions(
  children: ButtonGroupActionElement | ButtonGroupActionElement[],
): GetActionShape {
  const childArray = React.Children.toArray(
    children,
  ) as ButtonGroupActionElement[];

  let primaryActions = childArray.filter(
    action =>
      /* Checking the component type does not work in dev environments due to
       * wrapper components used for hot-reload
       * However, checking the type name does not work in prod due to code minification
       * Hence 2 different checks here */
      action.type === PrimaryAction || action.type.name === "PrimaryAction",
  ) as ButtonGroupPrimaryActionElement[];

  let secondaryActions = childArray.filter(
    action =>
      action.type === SecondaryAction || action.type.name === "SecondaryAction",
  );

  if (primaryActions.length > 2) {
    secondaryActions = primaryActions.slice(2).concat(secondaryActions);
    primaryActions = primaryActions.slice(0, 2);
  }

  if (primaryActions.length === 0) {
    primaryActions = secondaryActions.slice(0, 1);
    secondaryActions = secondaryActions.slice(1);
  }

  return { primaryActions, secondaryActions };
}
