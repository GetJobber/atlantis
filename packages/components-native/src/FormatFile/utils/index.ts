export { computeA11yLabel } from "./computeA11yLabel";
