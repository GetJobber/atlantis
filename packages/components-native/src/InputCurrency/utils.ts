export function countDecimal(value: number): number {
  const convertedValue = value.toString();

  if (convertedValue.includes(".")) {
    return convertedValue.split(".")[1].length;
  }

  return 0;
}

export function limitInputWholeDigits(
  value: number,
  maxInputLength: number,
): number {
  let convertedValue = value.toString();

  if (convertedValue.length > maxInputLength) {
    convertedValue = convertedValue.slice(0, maxInputLength);

    return parseFloat(convertedValue);
  }

  return value;
}

export function configureDecimal(
  decimalCount: number,
  maxDecimalPlaces: number,
  transformedValue: string,
  decimalPlaces: number,
): number {
  const targetDecimalPlaces = Math.min(
    Math.max(decimalCount, decimalPlaces),
    maxDecimalPlaces,
  );
  const precision = 10 ** targetDecimalPlaces;
  const convertedValue =
    Math.round(parseFloat(transformedValue) * precision) / precision;

  return convertedValue;
}

export function convertToNumber(value: string): string | number {
  const regexValidation = /^[0-9]*$/;

  if (value?.match?.(regexValidation)) {
    return parseFloat(value);
  }

  return value;
}

export const checkLastChar = (stringValue: string): boolean => {
  const lastChar = stringValue[stringValue.length - 1];

  return Boolean(Number(stringValue)) && lastChar !== "0" && lastChar !== ".";
};

export const isValidNumber = (
  numberedValue: string | number | undefined,
): boolean => {
  return (
    typeof numberedValue === "number" &&
    !isNaN(numberedValue) &&
    numberedValue !== 0
  );
};

export const getDecimalNumbers = (
  value: string,
  decimalSeparator: string,
): string => {
  const decimalValue = value.split(".")[1];

  if (!decimalValue) {
    return decimalSeparator;
  }

  return `${decimalSeparator}${decimalValue}`;
};

export const parseGivenInput = (
  value: string | undefined,
  decimalSeparator: string,
): [number, string | undefined, string] => {
  let decimalCount = 0;
  let decimalNumbers = "";
  let wholeIntegerValue = value;

  if (value?.includes(".")) {
    const splittedValue = value?.split(".");
    decimalCount = splittedValue[1].length;
    wholeIntegerValue = splittedValue[0];
    decimalNumbers = getDecimalNumbers(value, decimalSeparator);
  }

  return [decimalCount, wholeIntegerValue, decimalNumbers];
};

export const NUMBER_VALIDATION_REGEX =
  /^[-+]?(([0-9]*\.[0-9]+)|([0-9]+)|([0-9]+(\.?[0-9]+)?e[-+]?[0-9]+))$/;
