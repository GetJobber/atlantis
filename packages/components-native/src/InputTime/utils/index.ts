import { getTimezoneOffset } from "date-fns-tz";

export type MinutesIncrement = 15 | 30 | 60;

/**
 * Rounds up the time by increment.
 * - 15 mins - rounds to the next quarter time of `00:15`, `00:30`, `00:45`,
 *   and `01:00`
 * - 30 mins - rounds to the next half hour be it `00:30` or `01:00`
 * - 60 mins - rounds to the next hour. I.e., `02:01` gets rounded up
 *   to `03:00`.
 */
export function roundUpToNearestMinutes(
  date: Date,
  minutes: MinutesIncrement,
): Date {
  const ms = 1000 * 60 * minutes;

  return new Date(Math.ceil(date.getTime() / ms) * ms);
}

export function getTimeZoneOffsetInMinutes(
  timeZone: string,
  date?: Date,
): number {
  return getTimezoneOffset(timeZone, date) / 1000 / 60;
}
