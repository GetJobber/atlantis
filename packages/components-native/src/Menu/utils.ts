import { LayoutRectangle } from "react-native";
import { styles } from "./Menu.style";

interface ScreenInfo {
  windowHeight: number;
  headerHeight: number;
  windowWidth: number;
}

export function findViewpoint(
  screenInfo: ScreenInfo,
  activatorLayout: LayoutRectangle,
): { [key: string]: number | undefined } {
  const { windowHeight, windowWidth, headerHeight } = screenInfo;
  const pos: { [key: string]: number | undefined } = {};
  const menuWidth = styles.menu.width;
  const windowHalf = (windowHeight - headerHeight) / 2 + headerHeight;
  const menuPositionVertical =
    activatorLayout.y + activatorLayout.height > windowHalf
      ? "menuAbove"
      : "menuBelow";

  const menuPositionHorizontal =
    windowWidth / 2 > activatorLayout.width / 2 + activatorLayout.x
      ? "menuRight"
      : "menuLeft";

  const menuPadding = 36;

  getVerticalPosition(pos, windowHeight, activatorLayout, menuPositionVertical);

  getHorizontalPosition(
    pos,
    activatorLayout,
    windowWidth,
    menuPadding,
    menuWidth,
    menuPositionHorizontal,
  );

  return pos;
}

function getVerticalPosition(
  pos: { [key: string]: number | undefined },
  windowHeight: number,
  activatorLayout: LayoutRectangle,
  menuPositionVertical: string,
) {
  if (menuPositionVertical === "menuAbove") {
    getAbovePosition(pos, windowHeight, activatorLayout);
  }

  if (menuPositionVertical === "menuBelow") {
    getBelowPosition(pos, activatorLayout);
  }
}

function getBelowPosition(
  pos: { [key: string]: number | undefined },
  activatorLayout: LayoutRectangle,
) {
  pos.top = activatorLayout.y + activatorLayout.height;
}

function getAbovePosition(
  pos: { [key: string]: number | undefined },
  windowHeight: number,
  activatorLayout: LayoutRectangle,
) {
  pos.bottom = windowHeight - activatorLayout.y;
}

function getHorizontalPosition(
  pos: { [key: string]: number | undefined },
  activatorLayout: LayoutRectangle,
  windowWidth: number,
  menuPadding: number,
  menuWidth: number,
  menuPositionHorizontal: string,
) {
  if (menuPositionHorizontal === "menuRight") {
    getRightPosition(pos, activatorLayout, windowWidth, menuPadding, menuWidth);
  }

  if (menuPositionHorizontal === "menuLeft") {
    getLeftPosition(pos, activatorLayout, windowWidth, menuPadding, menuWidth);
  }
}

function getLeftPosition(
  pos: { [key: string]: number | undefined },
  activatorLayout: LayoutRectangle,
  windowWidth: number,
  menuHorizontalPadding: number,
  menuWidth: number,
) {
  const overflowLeft =
    windowWidth -
      activatorLayout.x -
      activatorLayout.width +
      activatorLayout.width / 2 -
      menuHorizontalPadding +
      menuWidth >
    windowWidth;

  const overflowRight =
    windowWidth -
      activatorLayout.x -
      activatorLayout.width +
      activatorLayout.width / 2 -
      menuHorizontalPadding <
    0;

  if (overflowLeft) {
    pos.right = undefined;
    pos.left = 0;
  } else if (overflowRight) {
    pos.right = 0;
  } else {
    pos.right = windowWidth - activatorLayout.x - activatorLayout.width;
  }
}

function getRightPosition(
  pos: { [key: string]: number | undefined },
  activatorLayout: LayoutRectangle,
  windowWidth: number,
  menuPadding: number,
  menuWidth: number,
) {
  const overflowRight =
    activatorLayout.x + activatorLayout.width / 2 - menuPadding + menuWidth >
    windowWidth;

  const overflowLeft =
    activatorLayout.x + activatorLayout.width / 2 - menuPadding < 0;

  if (overflowRight) {
    pos.left = undefined;
    pos.right = 0;
  } else if (overflowLeft) {
    pos.left = 0;
  } else {
    pos.left = activatorLayout.x;
  }
}
