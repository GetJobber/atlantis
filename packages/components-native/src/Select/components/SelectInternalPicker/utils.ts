import { Platform } from "react-native";
import {
  SelectInternalPickerProps,
  SelectOnOptionPressEvent,
} from "../../types";

export function handlePress(onChange: SelectInternalPickerProps["onChange"]) {
  return ({ nativeEvent: { event } }: SelectOnOptionPressEvent): void => {
    return onChange(event);
  };
}

export function isIOS14AndUp(): boolean {
  if (Platform.OS === "ios") {
    const majorVersionIOS = parseInt(Platform.Version, 10);

    return majorVersionIOS >= 14;
  }

  return false;
}
