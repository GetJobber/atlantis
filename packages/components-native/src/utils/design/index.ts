import { JobberStyle as mobileFoundationBase } from "@jobber/design/foundation.native";
import { Platform } from "react-native";

export const JobberStyle: typeof mobileFoundationBase = Platform.select({
  ios: () => require("@jobber/design/foundation.ios").JobberStyle,
  android: () => require("@jobber/design/foundation.android").JobberStyle,
  default: () => require("@jobber/design/foundation.native").JobberStyle,
})();
