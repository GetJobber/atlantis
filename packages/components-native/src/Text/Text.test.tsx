import React from "react";
import { render } from "@testing-library/react-native";
import { Text } from ".";
import { tokens } from "../utils/design";

it("renders text with no additional props", () => {
  const text = render(<Text>Test Text</Text>);
  expect(text.toJSON()).toMatchSnapshot();
});

it("renders text with base variation", () => {
  const text = render(<Text variation="base">Test Text</Text>);
  expect(text.toJSON()).toMatchSnapshot();
});

it("renders text with success variation", () => {
  const text = render(<Text variation="success">Test Text</Text>);
  expect(text.toJSON()).toMatchSnapshot();
});

it("renders text with error variation", () => {
  const text = render(<Text variation="error">Test Text</Text>);
  expect(text.toJSON()).toMatchSnapshot();
});

it("renders text with warn variation", () => {
  const text = render(<Text variation="warn">Test Text</Text>);
  expect(text.toJSON()).toMatchSnapshot();
});

it("renders text with info variation", () => {
  const text = render(<Text variation="info">Test Text</Text>);
  expect(text.toJSON()).toMatchSnapshot();
});

it("renders text with subdued variation", () => {
  const text = render(<Text variation="subdued">Test Text</Text>);
  expect(text.toJSON()).toMatchSnapshot();
});

it("renders text with success variation reverseTheme true", () => {
  const text = render(
    <Text variation="success" reverseTheme={true}>
      Test Text
    </Text>,
  );
  expect(text.toJSON()).toMatchSnapshot();
});

it("renders text with error variation reverseTheme true", () => {
  const text = render(
    <Text variation="error" reverseTheme={true}>
      Test Text
    </Text>,
  );
  expect(text.toJSON()).toMatchSnapshot();
});

it("renders text supporting with no additional props", () => {
  const text = render(<Text level="textSupporting">Test Text</Text>);
  expect(text.toJSON()).toMatchSnapshot();
});

it("renders text supporting with variation success", () => {
  const text = render(
    <Text level="textSupporting" variation="success">
      Test Text
    </Text>,
  );
  expect(text.toJSON()).toMatchSnapshot();
});

it("renders text supporting with variation success reverseTheme true", () => {
  const text = render(
    <Text level="textSupporting" variation="success" reverseTheme={true}>
      Test Text
    </Text>,
  );
  expect(text.toJSON()).toMatchSnapshot();
});

it("renders text with right alignment", () => {
  const text = render(<Text align="end">Test Text</Text>);
  expect(text.toJSON()).toMatchSnapshot();
});

it("renders text with center alignment", () => {
  const text = render(<Text align="center">Test Text</Text>);
  expect(text.toJSON()).toMatchSnapshot();
});

it("renders text with left alignment", () => {
  const text = render(<Text align="start">Test Text</Text>);
  expect(text.toJSON()).toMatchSnapshot();
});

it("renders text that is scaled down with adjustsFontSize true", () => {
  const text = render(
    <Text maxLines="base" adjustsFontSizeToFit={true}>
      The quick brown fox jumped over the lazy dog. The quick brown fox jumped
      over the lazy dog. The quick brown fox jumped over the lazy dog. The quick
      brown fox jumped over the lazy dog. The quick brown fox jumped over the
      lazy dog. The quick brown fox jumped over the lazy dog
    </Text>,
  );
  expect(text.toJSON()).toMatchSnapshot();
});

it("renders text that is not scaled down with adjustsFontSize false", () => {
  const text = render(
    <Text maxLines="base" adjustsFontSizeToFit={false}>
      The quick brown fox jumped over the lazy dog. The quick brown fox jumped
      over the lazy dog. The quick brown fox jumped over the lazy dog. The quick
      brown fox jumped over the lazy dog. The quick brown fox jumped over the
      lazy dog. The quick brown fox jumped over the lazy dog
    </Text>,
  );
  expect(text.toJSON()).toMatchSnapshot();
});

it("renders text with accessibilityRole set as text", () => {
  const text = render(<Text>Test Text</Text>);
  expect(text.getByRole("text")).toBeDefined();
});

it("renders with strikethrough styling", () => {
  const text = render(<Text strikeThrough={true}>Test Text</Text>);
  expect(text.toJSON()).toMatchSnapshot();
});

it("renders with italic styling", () => {
  const text = render(<Text italic>Test Text</Text>);
  expect(text.toJSON()).toMatchSnapshot();
});

it("renders text that is inaccessible", () => {
  const text = render(<Text hideFromScreenReader={true}>Test Text</Text>);
  expect(text.root.props).toEqual(
    expect.objectContaining({
      accessibilityRole: "none",
      accessible: false,
      importantForAccessibility: "no-hide-descendants",
    }),
  );
});

it("renders text with underline styling", () => {
  const text = render(<Text underline="dashed">Test Text</Text>);

  expect(text.toJSON()).toMatchSnapshot();
});

describe("UNSAFE_style", () => {
  it("applies custom styles via UNSAFE_style prop", () => {
    const customStyle = {
      textStyle: {
        fontSize: 20,
        color: tokens["color-blue--dark"],
      },
    };

    const { getByRole } = render(
      <Text UNSAFE_style={customStyle}>Test Text</Text>,
    );
    const textElement = getByRole("text");
    expect(textElement.props.style).toContainEqual(
      expect.objectContaining(customStyle.textStyle),
    );
  });
});
