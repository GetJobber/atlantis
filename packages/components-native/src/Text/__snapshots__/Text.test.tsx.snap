// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders text supporting with no additional props 1`] = `
<Text
  accessibilityRole="text"
  adjustsFontSizeToFit={false}
  allowFontScaling={true}
  collapsable={false}
  maxFontSizeMultiplier={1.1428571428571428}
  selectable={true}
  selectionColor="hsl(86, 100%, 46%)"
  style={
    [
      {
        "fontFamily": "inter-medium",
      },
      {
        "color": "hsl(198, 35%, 21%)",
      },
      {
        "textAlign": "left",
      },
      {
        "fontSize": 14,
        "lineHeight": 18,
      },
      {
        "letterSpacing": 0,
      },
    ]
  }
>
  Test Text
</Text>
`;

exports[`renders text supporting with variation success 1`] = `
<Text
  accessibilityRole="text"
  adjustsFontSizeToFit={false}
  allowFontScaling={true}
  collapsable={false}
  maxFontSizeMultiplier={1.1428571428571428}
  selectable={true}
  selectionColor="hsl(86, 100%, 46%)"
  style={
    [
      {
        "fontFamily": "inter-medium",
      },
      {
        "color": "hsl(107, 64%, 16%)",
      },
      {
        "textAlign": "left",
      },
      {
        "fontSize": 14,
        "lineHeight": 18,
      },
      {
        "letterSpacing": 0,
      },
    ]
  }
>
  Test Text
</Text>
`;

exports[`renders text supporting with variation success reverseTheme true 1`] = `
<Text
  accessibilityRole="text"
  adjustsFontSizeToFit={false}
  allowFontScaling={true}
  collapsable={false}
  maxFontSizeMultiplier={1.1428571428571428}
  selectable={true}
  selectionColor="hsl(86, 100%, 46%)"
  style={
    [
      {
        "fontFamily": "inter-medium",
      },
      {
        "color": "hsl(107, 58%, 33%)",
      },
      {
        "textAlign": "left",
      },
      {
        "fontSize": 14,
        "lineHeight": 18,
      },
      {
        "letterSpacing": 0,
      },
    ]
  }
>
  Test Text
</Text>
`;

exports[`renders text that is not scaled down with adjustsFontSize false 1`] = `
<Text
  accessibilityRole="text"
  adjustsFontSizeToFit={false}
  allowFontScaling={true}
  collapsable={false}
  maxFontSizeMultiplier={3.125}
  numberOfLines={4}
  selectable={true}
  selectionColor="hsl(86, 100%, 46%)"
  style={
    [
      {
        "fontFamily": "inter-regular",
      },
      {
        "color": "hsl(198, 35%, 21%)",
      },
      {
        "textAlign": "left",
      },
      {
        "fontSize": 16,
        "lineHeight": 20,
      },
      {
        "letterSpacing": 0,
      },
    ]
  }
>
  The quick brown fox jumped over the lazy dog. The quick brown fox jumped over the lazy dog. The quick brown fox jumped over the lazy dog. The quick brown fox jumped over the lazy dog. The quick brown fox jumped over the lazy dog. The quick brown fox jumped over the lazy dog
</Text>
`;

exports[`renders text that is scaled down with adjustsFontSize true 1`] = `
<Text
  accessibilityRole="text"
  adjustsFontSizeToFit={true}
  allowFontScaling={true}
  collapsable={false}
  maxFontSizeMultiplier={3.125}
  numberOfLines={4}
  selectable={true}
  selectionColor="hsl(86, 100%, 46%)"
  style={
    [
      {
        "fontFamily": "inter-regular",
      },
      {
        "color": "hsl(198, 35%, 21%)",
      },
      {
        "textAlign": "left",
      },
      {
        "fontSize": 16,
        "lineHeight": 20,
      },
      {
        "letterSpacing": 0,
      },
    ]
  }
>
  The quick brown fox jumped over the lazy dog. The quick brown fox jumped over the lazy dog. The quick brown fox jumped over the lazy dog. The quick brown fox jumped over the lazy dog. The quick brown fox jumped over the lazy dog. The quick brown fox jumped over the lazy dog
</Text>
`;

exports[`renders text with base variation 1`] = `
<Text
  accessibilityRole="text"
  adjustsFontSizeToFit={false}
  allowFontScaling={true}
  collapsable={false}
  maxFontSizeMultiplier={3.125}
  selectable={true}
  selectionColor="hsl(86, 100%, 46%)"
  style={
    [
      {
        "fontFamily": "inter-regular",
      },
      {
        "color": "hsl(198, 35%, 21%)",
      },
      {
        "textAlign": "left",
      },
      {
        "fontSize": 16,
        "lineHeight": 20,
      },
      {
        "letterSpacing": 0,
      },
    ]
  }
>
  Test Text
</Text>
`;

exports[`renders text with center alignment 1`] = `
<Text
  accessibilityRole="text"
  adjustsFontSizeToFit={false}
  allowFontScaling={true}
  collapsable={false}
  maxFontSizeMultiplier={3.125}
  selectable={true}
  selectionColor="hsl(86, 100%, 46%)"
  style={
    [
      {
        "fontFamily": "inter-regular",
      },
      {
        "color": "hsl(198, 35%, 21%)",
      },
      {
        "textAlign": "center",
      },
      {
        "fontSize": 16,
        "lineHeight": 20,
      },
      {
        "letterSpacing": 0,
      },
    ]
  }
>
  Test Text
</Text>
`;

exports[`renders text with error variation 1`] = `
<Text
  accessibilityRole="text"
  adjustsFontSizeToFit={false}
  allowFontScaling={true}
  collapsable={false}
  maxFontSizeMultiplier={3.125}
  selectable={true}
  selectionColor="hsl(86, 100%, 46%)"
  style={
    [
      {
        "fontFamily": "inter-regular",
      },
      {
        "color": "hsl(6, 64%, 51%)",
      },
      {
        "textAlign": "left",
      },
      {
        "fontSize": 16,
        "lineHeight": 20,
      },
      {
        "letterSpacing": 0,
      },
    ]
  }
>
  Test Text
</Text>
`;

exports[`renders text with error variation reverseTheme true 1`] = `
<Text
  accessibilityRole="text"
  adjustsFontSizeToFit={false}
  allowFontScaling={true}
  collapsable={false}
  maxFontSizeMultiplier={3.125}
  selectable={true}
  selectionColor="hsl(86, 100%, 46%)"
  style={
    [
      {
        "fontFamily": "inter-regular",
      },
      {
        "color": "hsl(6, 64%, 51%)",
      },
      {
        "textAlign": "left",
      },
      {
        "fontSize": 16,
        "lineHeight": 20,
      },
      {
        "letterSpacing": 0,
      },
    ]
  }
>
  Test Text
</Text>
`;

exports[`renders text with info variation 1`] = `
<Text
  accessibilityRole="text"
  adjustsFontSizeToFit={false}
  allowFontScaling={true}
  collapsable={false}
  maxFontSizeMultiplier={3.125}
  selectable={true}
  selectionColor="hsl(86, 100%, 46%)"
  style={
    [
      {
        "fontFamily": "inter-regular",
      },
      {
        "color": "hsl(207, 61%, 34%)",
      },
      {
        "textAlign": "left",
      },
      {
        "fontSize": 16,
        "lineHeight": 20,
      },
      {
        "letterSpacing": 0,
      },
    ]
  }
>
  Test Text
</Text>
`;

exports[`renders text with left alignment 1`] = `
<Text
  accessibilityRole="text"
  adjustsFontSizeToFit={false}
  allowFontScaling={true}
  collapsable={false}
  maxFontSizeMultiplier={3.125}
  selectable={true}
  selectionColor="hsl(86, 100%, 46%)"
  style={
    [
      {
        "fontFamily": "inter-regular",
      },
      {
        "color": "hsl(198, 35%, 21%)",
      },
      {
        "textAlign": "left",
      },
      {
        "fontSize": 16,
        "lineHeight": 20,
      },
      {
        "letterSpacing": 0,
      },
    ]
  }
>
  Test Text
</Text>
`;

exports[`renders text with no additional props 1`] = `
<Text
  accessibilityRole="text"
  adjustsFontSizeToFit={false}
  allowFontScaling={true}
  collapsable={false}
  maxFontSizeMultiplier={3.125}
  selectable={true}
  selectionColor="hsl(86, 100%, 46%)"
  style={
    [
      {
        "fontFamily": "inter-regular",
      },
      {
        "color": "hsl(198, 35%, 21%)",
      },
      {
        "textAlign": "left",
      },
      {
        "fontSize": 16,
        "lineHeight": 20,
      },
      {
        "letterSpacing": 0,
      },
    ]
  }
>
  Test Text
</Text>
`;

exports[`renders text with right alignment 1`] = `
<Text
  accessibilityRole="text"
  adjustsFontSizeToFit={false}
  allowFontScaling={true}
  collapsable={false}
  maxFontSizeMultiplier={3.125}
  selectable={true}
  selectionColor="hsl(86, 100%, 46%)"
  style={
    [
      {
        "fontFamily": "inter-regular",
      },
      {
        "color": "hsl(198, 35%, 21%)",
      },
      {
        "textAlign": "right",
      },
      {
        "fontSize": 16,
        "lineHeight": 20,
      },
      {
        "letterSpacing": 0,
      },
    ]
  }
>
  Test Text
</Text>
`;

exports[`renders text with subdued variation 1`] = `
<Text
  accessibilityRole="text"
  adjustsFontSizeToFit={false}
  allowFontScaling={true}
  collapsable={false}
  maxFontSizeMultiplier={3.125}
  selectable={true}
  selectionColor="hsl(86, 100%, 46%)"
  style={
    [
      {
        "fontFamily": "inter-regular",
      },
      {
        "color": "hsl(197, 15%, 43%)",
      },
      {
        "textAlign": "left",
      },
      {
        "fontSize": 16,
        "lineHeight": 20,
      },
      {
        "letterSpacing": 0,
      },
    ]
  }
>
  Test Text
</Text>
`;

exports[`renders text with success variation 1`] = `
<Text
  accessibilityRole="text"
  adjustsFontSizeToFit={false}
  allowFontScaling={true}
  collapsable={false}
  maxFontSizeMultiplier={3.125}
  selectable={true}
  selectionColor="hsl(86, 100%, 46%)"
  style={
    [
      {
        "fontFamily": "inter-regular",
      },
      {
        "color": "hsl(107, 64%, 16%)",
      },
      {
        "textAlign": "left",
      },
      {
        "fontSize": 16,
        "lineHeight": 20,
      },
      {
        "letterSpacing": 0,
      },
    ]
  }
>
  Test Text
</Text>
`;

exports[`renders text with success variation reverseTheme true 1`] = `
<Text
  accessibilityRole="text"
  adjustsFontSizeToFit={false}
  allowFontScaling={true}
  collapsable={false}
  maxFontSizeMultiplier={3.125}
  selectable={true}
  selectionColor="hsl(86, 100%, 46%)"
  style={
    [
      {
        "fontFamily": "inter-regular",
      },
      {
        "color": "hsl(107, 58%, 33%)",
      },
      {
        "textAlign": "left",
      },
      {
        "fontSize": 16,
        "lineHeight": 20,
      },
      {
        "letterSpacing": 0,
      },
    ]
  }
>
  Test Text
</Text>
`;

exports[`renders text with underline styling 1`] = `
<Text
  accessibilityRole="text"
  adjustsFontSizeToFit={false}
  allowFontScaling={true}
  collapsable={false}
  maxFontSizeMultiplier={3.125}
  selectable={true}
  selectionColor="hsl(86, 100%, 46%)"
  style={
    [
      {
        "fontFamily": "inter-regular",
      },
      {
        "color": "hsl(198, 35%, 21%)",
      },
      {
        "textAlign": "left",
      },
      {
        "fontSize": 16,
        "lineHeight": 20,
      },
      {
        "letterSpacing": 0,
      },
      {
        "textDecorationStyle": "dotted",
      },
      {
        "textDecorationColor": "hsl(197, 15%, 43%)",
        "textDecorationLine": "underline",
      },
    ]
  }
>
  Test Text
</Text>
`;

exports[`renders text with warn variation 1`] = `
<Text
  accessibilityRole="text"
  adjustsFontSizeToFit={false}
  allowFontScaling={true}
  collapsable={false}
  maxFontSizeMultiplier={3.125}
  selectable={true}
  selectionColor="hsl(86, 100%, 46%)"
  style={
    [
      {
        "fontFamily": "inter-regular",
      },
      {
        "color": "hsl(51, 64%, 24%)",
      },
      {
        "textAlign": "left",
      },
      {
        "fontSize": 16,
        "lineHeight": 20,
      },
      {
        "letterSpacing": 0,
      },
    ]
  }
>
  Test Text
</Text>
`;

exports[`renders with italic styling 1`] = `
<Text
  accessibilityRole="text"
  adjustsFontSizeToFit={false}
  allowFontScaling={true}
  collapsable={false}
  maxFontSizeMultiplier={3.125}
  selectable={true}
  selectionColor="hsl(86, 100%, 46%)"
  style={
    [
      {
        "fontFamily": "inter-italic",
      },
      {
        "color": "hsl(198, 35%, 21%)",
      },
      {
        "textAlign": "left",
      },
      {
        "fontSize": 16,
        "lineHeight": 20,
      },
      {
        "letterSpacing": 0,
      },
      {
        "fontStyle": "italic",
      },
    ]
  }
>
  Test Text
</Text>
`;

exports[`renders with strikethrough styling 1`] = `
<Text
  accessibilityRole="text"
  adjustsFontSizeToFit={false}
  allowFontScaling={true}
  collapsable={false}
  maxFontSizeMultiplier={3.125}
  selectable={true}
  selectionColor="hsl(86, 100%, 46%)"
  style={
    [
      {
        "fontFamily": "inter-regular",
      },
      {
        "color": "hsl(198, 35%, 21%)",
      },
      {
        "textAlign": "left",
      },
      {
        "fontSize": 16,
        "lineHeight": 20,
      },
      {
        "letterSpacing": 0,
      },
      {
        "textDecorationLine": "line-through",
      },
    ]
  }
>
  Test Text
</Text>
`;

exports[`supports nested Text children with mixed styles 1`] = `
<Text
  accessibilityRole="text"
  adjustsFontSizeToFit={false}
  allowFontScaling={true}
  collapsable={false}
  maxFontSizeMultiplier={3.125}
  selectable={true}
  selectionColor="hsl(86, 100%, 46%)"
  style={
    [
      {
        "fontFamily": "inter-regular",
      },
      {
        "color": "hsl(198, 35%, 21%)",
      },
      {
        "textAlign": "left",
      },
      {
        "fontSize": 16,
        "lineHeight": 20,
      },
      {
        "letterSpacing": 0,
      },
    ]
  }
>
  Hello 
  <Text
    accessibilityRole="text"
    adjustsFontSizeToFit={false}
    allowFontScaling={true}
    collapsable={false}
    maxFontSizeMultiplier={3.125}
    selectable={true}
    selectionColor="hsl(86, 100%, 46%)"
    style={
      [
        {
          "fontFamily": "inter-semibold",
        },
        {
          "color": "hsl(198, 35%, 21%)",
        },
        {
          "textAlign": "left",
        },
        {
          "fontSize": 16,
          "lineHeight": 20,
        },
        {
          "letterSpacing": 0,
        },
      ]
    }
  >
    World
  </Text>
  !
</Text>
`;
